dot-net/Centroid/Config.cs

```diff
@@ -78,7 +78,23 @@ public override bool TryConvert(ConvertBinder binder, out object result)
 
             try
             {
-                result = Convert.ChangeType(RawConfig, binder.Type);
+                JToken rawConfig = RawConfig;
+                result = rawConfig.ToObject(binder.Type);
+                //ensure at least one property in the config bound to the type
```

Comment:
I guess the question we should answer is: do we need to police this?  Should we just be happy if it completes without throwing?

If I use the `JsonConvert` Serialize, then Deserialize method of conversion, then I'll end up with a JObject with just default values if no properties match.